docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1174 @@
 JSON functions and operators
 ============================
 
+The SQL standard describes functions and operators to process JSON data. They
+allow you to access JSON data according to its structure, generate JSON data,
+and store it persistently in SQL tables.
+
+Importantly, the SQL standard imposes that there is no dedicated data type to
+represent JSON data in SQL. Instead, JSON data is represented as character or
+binary strings. Although Trino supports ``JSON`` type, it is not used or
+produced by the following functions.
+
+Trino supports three functions for querying JSON data:
+:ref:`json_exists<json_exists>`,
+:ref:`json_query<json_query>`, and :ref:`json_value<json_value>`. Each of them
+is based on the same mechanism of exploring and processing JSON input using
+JSON path.
+
+JSON path language
+------------------
+
+The JSON path language is a special language, used exclusively by certain SQL
+operators to specify the query to perform on the JSON input. Although JSON path
+expressions are embedded in an SQL query, their syntax significantly differs
+from SQL. The semantics of predicates, operators, etc. in JSON path expressions
+generally follow the semantics of SQL. The JSON path language is case-sensitive
+for keywords and identifiers.
+
+.. _json_path_syntax_and_semantics:
+
+JSON path syntax and semantics
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A JSON path expression, similar to an SQL expression, is a recursive structure.
+Although the name "path" suggests a linear sequence of operations going step by
+step deeper into the JSON structure, a JSON path expression is in fact a tree.
+It can access the input JSON item multiple times, in multiple ways, and combine
+the results. Moreover, the result of a JSON path expression is not a single
+item, but an ordered sequence of items. Each of the sub-expressions takes one
+or more input sequences, and returns a sequence as the result.
+
+.. note::
+
+    In the lax mode, most path operations first unnest all JSON arrays in the
+    input sequence. Any divergence from this rule is mentioned in the following
+    listing. Path modes are explained in :ref:`json_path_modes`.
+
+The JSON path language supports the following features:
+
+literals
+''''''''
+
+- numeric literals
+
+  Their semantics is the same as in SQL.
```

Comment:
"semantics *are* the same" (assuming we're talking about multiple behaviors that can vary for different types). It would be clearer to describe what we're talking about, though, as just saying "semantics" is too vague. What specific behaviors are similar to SQL?

Reply:
reworded